src/main/java/org/breedinginsight/services/geno/impl/GigwaGenoServiceImpl.java

```diff
@@ -0,0 +1,732 @@
+                executor.execute(() -> {
+                    try {
+                        processSubmission(gigwaAuthToken, program, experimentId, fileContents, uploadedFile.getFilename(), upload, progress);
+                    } catch (Exception ignored) {
+                    }
+                });
+            }
+        } catch (InstantiationException | IOException e) {
+            progress.setStatuscode((short) HttpStatus.INTERNAL_SERVER_ERROR.getCode());
+            progress.setMessage("An error occurred while trying to validate sample information");
+            importDAO.updateProgress(progress);
+            throw new RuntimeException(e);
+        }
+
+        ImportResponse response = new ImportResponse();
+        response.setImportId(upload.getId());
+        response.setProgress(progress);
+        return response;
+    }
+
+    private boolean validateSamples(Program program, UUID experimentId, byte[] fileContents, ImportUpload upload, String gigwaAuthToken) throws DoesNotExistException, ApiException, InstantiationException {
+        log.debug("Validating samples in submitted VCF file for experiment: " + experimentId);
+
+        BrAPIClient brAPIClient = programDAO.getCoreClient(program.getId());
+        brAPIClient.setBasePath(gigwaHost + GIGWA_BRAPI_BASE_PATH);
+        Authentication authorizationToken = brAPIClient.getAuthentication("AuthorizationToken");
+        if(authorizationToken instanceof OAuth) {
+            ((OAuth)authorizationToken).setAccessToken(gigwaAuthToken);
+        }
+        BrAPIClient brapiPhenoClient = programDAO.getPhenoClient(program.getId());
+
+        Set<String> obsUnitNames = fetchObservationUnits(brapiPhenoClient, experimentId).stream().map(ou -> Utilities.removeProgramKeyAndUnknownAdditionalData(ou.getObservationUnitName(), program.getKey())).collect(Collectors.toSet());
+
+        log.debug("searching for the VCF header row");
+        String[] headerParts = null;
+        Scanner sc = new Scanner(new ByteArrayInputStream(fileContents), "UTF-8");
+        boolean foundHeader = false;
+        while (sc.hasNextLine() && !foundHeader) {
+            String line = sc.nextLine();
+            if(line.startsWith("#CHROM")) {
+                log.debug("Header row found! -> " + line);
+                foundHeader = true;
+                headerParts = line.split("\t");
+            }
+        }
+
+        if(!foundHeader) {
+            upload.getProgress().setStatuscode((short)HttpStatus.BAD_REQUEST.getCode());
+            upload.getProgress().setMessage("Could not find header row in file");
+            importDAO.updateProgress(upload.getProgress());
+            return false;
+        }
+
+        List<String> samples = new ArrayList<>();
+        boolean validHeader = false;
+        if(headerParts.length >= 8) {
+            validHeader = validateVcfHeader(headerParts);
+            if(validHeader) {
+                validHeader = true;
```

Comment:
You may no longer need` validHeader = true;`.  validHeader must already be true to get here.

Reply:
Removed